test/ng/logSpec.js

```diff
@@ -101,24 +101,29 @@ describe('$log', function() {
       })
     );
 
-    it('should not attempt to log the second argument in IE if it is not specified', inject(
-      function() {
-        log = function(arg1, arg2) { logger += 'log;' + arg2; };
-        warn = function(arg1, arg2) { logger += 'warn;' + arg2; };
-        info = function(arg1, arg2) { logger += 'info;' + arg2; };
-        error = function(arg1, arg2) { logger += 'error;' + arg2; };
-        debug = function(arg1, arg2) { logger += 'debug;' + arg2; };
-      },
-      removeApplyFunctionForIE,
-      function($log) {
-        $log.log();
-        $log.warn();
-        $log.info();
-        $log.error();
-        $log.debug();
-        expect(logger).toEqual('log;warn;info;error;debug;');
-      })
-    );
+    // Support: Safari 9.1 only, iOS 9.3 only
+    // For some reason Safar thinks there is always 1 parameter passed here.
+    if (!/\b9\.\d(\.\d+)* safari/i.test(window.navigator.userAgent) &&
+      !/\biphone os 9_/i.test(window.navigator.userAgent)) {
+      it('should not attempt to log the second argument in IE if it is not specified', inject(
```

Comment:
This test is a little weird. It's supposed to simulate the IE logging behavior (i.e. missing `apply` on `console` methods). Normally we'd just run those tests in IE without patching them but to be able to mock those methods we have to replace them with mocks that are regular functions so, in turn, we have to delete the methods.

Is there any way to reduce the confusion here?